Index test in generated animation list apply functions is wrong way round

anttijk opened the pull request:
#### 3d1b2ebb18804a2bec7bd17d2a78cad0bfc9197d
<pre>
Index test in generated animation list apply functions is wrong way round
<a href="https://bugs.webkit.org/show_bug.cgi?id=265124">https://bugs.webkit.org/show_bug.cgi?id=265124</a>
<a href="https://rdar.apple.com/118644028">rdar://118644028</a>

Reviewed by Tim Nguyen.

* Source/WebCore/css/process-css-properties.py:

We are adding unnecessary entries to the list and then later trimming them.

Canonical link: <a href="https://commits.webkit.org/270973@main">https://commits.webkit.org/270973@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/56ccfa42bf6c500e802b4ab8f22bb5def936e34f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5568 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28188 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29163 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24642 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27398 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7425 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2960 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24513 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27214 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4394 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23138 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3843 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3918 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24136 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29798 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24599 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24543 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30128 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3931 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2129 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28049 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5390 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6473 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4397 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4300 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -3361,7 +3361,7 @@ def _generate_animation_property_value_setter(self, to, property):
         to.write(f"    // Walk each value and put it into an animation, creating new animations as needed.")
         to.write(f"    for (auto& currentValue : downcast<CSSValueList>(value)) {{")
-        to.write(f"        if (childIndex <= list.size())")
+        to.write(f"        if (childIndex >= list.size())")
```
Can we make this pattern consistent with https://searchfox.org/wubkat/rev/114aa2c8594807cbc34dd5dff48a9e2addfd1e00/Source/WebCore/css/process-css-properties.py#3350 ? (no specific preference on the exact style, up to you)

anttijk replied:
I'll just fix the bug, there are all sorts of cleanups that could be done here, better handled separately.

anttijk committed:
Index test in generated animation list apply functions is wrong way round
https://bugs.webkit.org/show_bug.cgi?id=265124
rdar://118644028

Reviewed by Tim Nguyen.

* Source/WebCore/css/process-css-properties.py:

We are adding unnecessary entries to the list and then later trimming them.

Canonical link: https://commits.webkit.org/270973@main
